src/components/Dasboard/Cards/DasboardCards.tsx

```diff
@@ -23,15 +78,185 @@ export default function DashboardCard({
 			marginTop: theme.spacing(0),
 		},
 	}));
+	const [currentFilter, setCurrentFilter] = useState<any>();
+
+	/*setting first filter as default filter*/
+	useEffect(() => {
+		if (cardFilter && cardFilter.length) {
+			setCurrentFilter(cardFilter[0]);
+		}
+	}, []);
 
 	const classes = useStyles();
+	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+	const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+		setAnchorEl(event.currentTarget);
+	};
+	const handleClose = () => {
+		setAnchorEl(null);
+	};
+	const [progressDialogOpen, setProgressDialogOpen] = React.useState(false);
 	return (
 		<Card raised={false} className={classes.card} style={{ height: cardHeight }}>
 			<CardContent>
-				<Typography color="primary" gutterBottom>
-					{title}
-				</Typography>
-				{children}
+				<Grid container>
+					<Grid item md={12} container justify="space-between">
+						{title && (
+							<Box mt={1} mb={1}>
+								<Typography color="primary" gutterBottom>
+									{title}
+								</Typography>
+							</Box>
+						)}
+						{cardFilter && cardFilter.length > 0 && (
+							<>
+								<IconButton onClick={handleClick}>
+									<FilterListIcon fontSize="small" />
+								</IconButton>
+
+								<Menu
+									id="simple-menu-budget-org"
+									anchorEl={anchorEl}
+									keepMounted
+									open={Boolean(anchorEl)}
+									onClose={handleClose}
+								>
+									{cardFilter.map(
+										(
+											filter: { label: string; filter: object },
+											mapIndex: number
+										) => {
+											return (
+												<MenuItem
+													onClick={() => {
+														setCurrentFilter(filter);
+														handleClose();
+													}}
+												>
+													{filter.label}
+												</MenuItem>
+											);
+										}
+									)}
+								</Menu>
+							</>
+						)}
+					</Grid>
+					{props.type === CARD_TYPES.PROGRESS && (
```

Comment:
Can we handle this outside of return funcction. It will get more complicated if more types are added.

Reply:
Doing it in another PR